--- ./configure.in.org	2008-04-19 01:30:14.000000000 +0200
+++ ./configure.in	2008-04-23 01:40:44.989833438 +0200
@@ -64,7 +64,7 @@ AC_SUBST(GETTEXT_PACKAGE)
 AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE,"$GETTEXT_PACKAGE", [Gettext package.])
 
 dnl Add the languages which your application supports here.
-ALL_LINGUAS="cz da de en fr it nb pl ru fi es"
+ALL_LINGUAS="cs da de en fr it nb pl ru fi es"
 AM_GLIB_GNU_GETTEXT
 AC_CHECK_FUNCS(memmem)
 
