--- rawstudio-2.0/plugins/load-gdk/exiv2-colorspace.cpp.orig	2012-02-13 00:47:29.504505358 +0100
+++ rawstudio-2.0/plugins/load-gdk/exiv2-colorspace.cpp	2012-02-13 00:48:32.694945836 +0100
@@ -125,11 +125,11 @@
 
 					int compression_type;
 					/* Extract embedded ICC profile */
-					if (info_ptr->valid & PNG_INFO_iCCP)
+					if (png_get_valid(png_ptr, info_ptr, PNG_INFO_iCCP))
 					{
 						png_uint_32 retval = png_get_iCCP (png_ptr, info_ptr,
 													(png_charpp) &icc_profile_title, &compression_type,
-													(png_charpp) &icc_profile, (png_uint_32*) &icc_profile_size);
+													(png_bytepp) &icc_profile, (png_uint_32*) &icc_profile_size);
 						if (retval != 0)
 						{
 							RSIccProfile *icc = rs_icc_profile_new_from_memory((gchar*)icc_profile, icc_profile_size, TRUE);
--- rawstudio-2.0/plugins/output-pngfile/output-pngfile.c~	2011-03-26 02:52:29.000000000 +0100
+++ rawstudio-2.0/plugins/output-pngfile/output-pngfile.c	2012-02-13 00:51:07.129346358 +0100
@@ -23,6 +23,7 @@
 #include <gettext.h>
 #include "config.h"
 #include <png.h>
+#include <zlib.h>
 
 #define RS_TYPE_PNGFILE (rs_pngfile_type)
 #define RS_PNGFILE(obj) (G_TYPE_CHECK_INSTANCE_CAST ((obj), RS_TYPE_PNGFILE, RSPngfile))
